python/pyhdk/hdk.py

```diff
@@ -0,0 +1,1348 @@
+        >>>
+        >>> test = hdk.import_csv("test.csv")
+        >>> res = hdk.sql("SELCT type, count(*) FROM test GROUP BY type;", test=test)
+        """
+        ra = self._calcite.process(sql_query)
+        ra_executor = RelAlgExecutor(self._executor, self._storage, self._data_mgr, ra)
+        return ra_executor.execute()
+
+    @not_implemented
+    def scan(self, table_name):
+        """
+        Create a scan query node referencing specified table.
+
+        Parameters
+        ----------
+        table_name : str
+            A name of a referenced table.
+
+        Returns
+        -------
+        QueryNode
+        """
+        pass
+
+    @not_implemented
+    def type(self, type_str):
+        """
+        Parse string type representation into TypeInfo object.
+
+        Parameters
+        ----------
+        type_str : str
+            String type representation. Type strings are case insensitive. It usually
+            consists of type name, optional length in bits, optional parameter,
+            optional unit parameter, and optional nullability suffix.
+
+            By default types are nullable, "[nn]" suffix is used for non-nullable types.
+
+            Datetime types support time unit specification. Supported suffixes:
+            [m] = month, [d] = day, [s] = second, [ms] = millisecond, [us] = microsecond,
+            [ns] = nanosecond.
+
+            Supported types:
+
+            Null type: "nullt"
+
+            Integer type examples: "int8", "int16", "int32[nn]", "int64", "int" (equal
+            to "int64")
+
+            Floating point type examples: "fp32", "fp64[nn]", "fp" (equal to "fp64")
+
+            Decimal type examples: "dec(10, 2)" (equal to "dec64(10, 2)"), "dec64(12, 1)",
+            "DECIMAL64(10, 2)[NN]"
+
+            Boolean type examples: "bool", "bool[nn]"
+
+            Varchar type examples: "varcahr(10)[nn]", "varchar(0)"
+
+            Text type examples: "text", "TEXT[NN]"
+
+            Date type examples: "date32" (equal to "date32[s]"), "date64[s][nn]",
+            "date" (equal to "date64[s]")
+
+            Time type examples: "time16[s]", "time32[ms]", "time" (equal to "time64[us]")
+
+            Timestamp type examples: "timestamp64[s]", "timestamp" (equal to "timestamp64[ms]")
+
+            Interval type examples: "interval32[ms]", "interval" (equal to "interval64[ms])
+
+            Dictionary type exampless: "dict32(text[nn])[10]" (10 - dictionary ID), "dict"
+            (equal to "dict32(text)[0]), "dict16[5]" (equal to "dict16(text)[5]")
+
+            Fixed length array type examples: array(int)(2), "array(fp64[nn])(8)[nn]",
+            "array(dict(text[nn])[10])(4)[nn]"
+
+            Variable length array type examples: "array(int)", "array(fp64[nn])[nn]"
+
+        Returns
+        -------
+        TypeInfo
+        """
+        pass
+
+    @not_implemented
+    def cst(self, value, type=None, scale_decimal=True):
```

Comment:
`cst` looks more like a cast, not constant. Do we really want to save 2 letters in a very clear `const`?

Reply:
It is a creation of an expression, not simply a cast. E. g. `proj("a")` and `proj(hdk.cst("a"))` are completely different things.  `type` parameter allows us to avoid additional cast. `hdk.cst("2020-10-05", "date")` is equal to `hdk.cst("2020-10-05").cast("date")`.

I'm used to having `const` as a reserved word. But we can use it here I guess or create an alias to have both.

Reply:
What about `.literal()`?

Reply:
I'm not sure about introducing another alias for constants. Representing expression is `hdk::ir::Constant`, so it makes sense to use similar words in API.

Reply:
Yeah, I would probably spell it out then. Worth the extra typing. :)